Kinvey-Xamarin/Realtime/Stream.cs

```diff
@@ -0,0 +1,292 @@
+		public Stream(string streamName, AbstractClient client = null)
+		{
+			StreamName = streamName;
+
+			KinveyClient = client ?? Client.SharedClient;
+		}
+
+		/// <summary>
+		/// Grant access to the stream for publishing and/or subscribing to specified users and/or user groups.
+		/// </summary>
+		/// <returns>True if the grant request was successful, otherwise false.</returns>
+		/// <param name="streamACL"><see cref="StreamAccessControlList"/> object used to determine which users should be granted access for publish and/or subscribe access.</param>
+		/// <param name="ct">[optional] CancellationToken used to cancel the request.</param>
+		public async Task<bool> GrantStreamAccess(StreamAccessControlList streamACL, CancellationToken ct = default(CancellationToken))
+		{
+			GrantStreamAccessRequest requestGrant = BuildGrantStreamAccessRequest(streamACL);
+			ct.ThrowIfCancellationRequested();
+			var success = await requestGrant.ExecuteAsync();
+			return (success != null);
+		}
+
+		/// <summary>
+		/// Publish a message of type {T} to the specified user.
+		/// </summary>
+		/// <param name="receiverID">Receiver identifier.</param>
+		/// <param name="message">Message.</param>
+		public async Task<bool> Publish(string receiverID, T message)
+		{
+			bool result = false;
+
+			// Make KCS request for publish access for the given receiverID
+			// KCS will return, if successful, a response which will include the PubNub channel name
+			JObject response = await RequestPublishAccess(receiverID);
+
+			if (response != null)
+			{
+				string publishChannel = response[Constants.STR_REALTIME_PUBLISH_SUBSTREAM_CHANNEL_NAME].ToString();
+				result = RealtimeRouter.Publish(publishChannel, receiverID, message);
+			}
+
+			return result;
+		}
+
+		/// <summary>
+		/// Subscribe the specified callback.
+		/// </summary>
+		/// <param name="subscribeID">The ID of the user to subscribe to.</param>
+		/// <param name="callback">Callback used to forward realtime messages.</param>
+		public async Task<bool> Subscribe(string subscribeID, KinveyRealtimeDelegate<T> callback)
+		{
+			bool success = false;
+
+			// Make KCS request for subscribe access to a user with the given subscribeID
+			success = await RequestSubscribeAccess(subscribeID);
+
+			if (success && callback != null)
+			{
+				RealtimeCallback = callback;
+
+				routerCallback = new KinveyRealtimeDelegate<string>
+				{
+					onError = (error) => RealtimeCallback.onError(error),
+					onSuccess = (message) => {
+						var messageObj = Newtonsoft.Json.JsonConvert.DeserializeObject<T>(message);
+						RealtimeCallback.onSuccess(messageObj);
+					},
+					OnStatus = (status) => RealtimeCallback.OnStatus(status)
+				};
+
+				RealtimeRouter.SubscribeStream(StreamName, routerCallback);
+				success = true;
+			}
+
+			return success;
+		}
+
+		/// <summary>
+		/// Unsubscribe this instance.
+		/// </summary>
+		public async Task Unsubscribe(string subscribeID)
```

Comment:
Will fix this to make sure we uninitialize the `RealtimeRouter` on every `Logout()` call.